Add warning regarding github vcpkg binary cache removal (#478)
microsoft/vcpkg-docs: vcpkg/consume/binary-caching-github-actions-cache.md

---
# Tutorial: Set up a vcpkg binary cache using GitHub Actions Cache

> > [!WARNING]

Reviewer: ```suggestion
> [!CAUTION]
```

> > Vcpkg binary cache for GitHub Actions is **deprecated** and this tutorial is **no longer maintained**.  
> > For more information and migration guidance, please see [GitHub PR #1662](https://github.com/microsoft/vcpkg-tool/pull/1662).

Reviewer: ```suggestion
> The GitHub Actions Cache backend for binary caching has been removed. This tutorial is no longer maintained.  
> For more information and migration guidance, please see [GitHub PR #1662](https://github.com/microsoft/vcpkg-tool/pull/1662).
```
